programs/pkey/pk_decrypt.c

```diff
@@ -132,27 +133,29 @@ int main( int argc, char *argv[] )
     mbedtls_printf( "\n  . Decrypting the encrypted data" );
     fflush( stdout );
 
-    if( ( ret = mbedtls_pk_decrypt( &pk, buf, i, result, &olen, sizeof(result),
-                            mbedtls_ctr_drbg_random, &ctr_drbg ) ) != 0 )
+    if( ( ret = mbedtls_pk_decrypt( &pk, buf, i, result, &olen,
+                                    sizeof( result ), mbedtls_ctr_drbg_random,
+                                    &ctr_drbg ) ) != 0 )
     {
-        mbedtls_printf( " failed\n  ! mbedtls_pk_decrypt returned -0x%04x\n", -ret );
+        mbedtls_printf( " failed\n  ! mbedtls_pk_decrypt returned -0x%04x\n",
+                        -ret );
         goto exit;
     }
 
     mbedtls_printf( "\n  . OK\n\n" );
 
     mbedtls_printf( "The decrypted result is: '%s'\n\n", result );
-
     ret = 0;
 
 exit:
+    mbedtls_pk_free( &pk );
```

Comment:
As a general rule, I always prefer having the de-initialization\free etc.. function be in reverse order than they were initiated\allocated.  This avoids issues of dependencies between objects that were initiated\allocated before. There is no effect in this specific use case though.

Reply:
@RonEld Thanks for pointing that out! I agree that this is a valuable rule to adhere to.